### PR TITLE
Approval checking unit tests

### PR DESCRIPTION
**Pull request by Lldenaurois**
This PR intends to add unit tests for approval checking in the Approval Voting subsystem

### CODE REVIEW

File: node/core/approval-voting/src/approval_checking.rs [hunk 1]
```diff
@@ -938,6 +948,48 @@ mod tests {
...
+	fn filled_tranche_iterator_yields_sequential_tranches() {
+		const PREFIX: u32 = 10;
+
+		let test_tranches = vec![
+			vec![],	 // empty set
+			vec![0],	// zero start
+			vec![0, 3], // zero start with gap
+			vec![2],	// non-zero start
+			vec![2, 4], // non-zero start with gap
+		];
```
Comment (rphmeier) [inline node/core/approval-voting/src/approval_checking.rs:962]: Some test cases with longer gaps and longer sequential runs would be good

Comment (Lldenaurois) [reply node/core/approval-voting/src/approval_checking.rs:962]: Yes, the goal here was to get started with something and see whether it's moving in the right direction. I will continue to expand on this.

**Comment by rphmeier** [comment]
Let's get the bugfix merged in this PR and then you can continue adding tests in a further branch

File: node/core/approval-voting/src/approval_checking.rs (hunk 1 above)
Comment (rphmeier) [inline node/core/approval-voting/src/approval_checking.rs:962]:
```suggestion
			vec![0, 1, 2], // zero start with run and no gap
			vec![2, 3, 4, 8], // non-zero start with run and gap
			vec![0, 1, 2, 5, 6, 7], // zero start with runs and gap
		];
```

**Comment by Lldenaurois** [comment]
Should be ready to go.

I've opened a new PR with additional tests. I'll build on top of that one to expand on the testing for this subsystem.

**Comment by rphmeier** [comment]
This needs a merge with master and then should be good to go.

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
bot merge

**Pull request merged**